src/common/Square.tsx

```diff
@@ -1,22 +1,29 @@
 import styled from "styled-components";
-import { SQUARE_USAGE } from "../utils/mixins";
+import React from "react";
 
-const StyledWeek = styled.div<{ squareColor: string }>`
-    width: 12px;
-    height: 12px;
+const StyledSquare = styled.button<{ $color: string }>`
     border: 1px solid black;
-    background-color: ${({ squareColor }) => squareColor};
+    width: 14px;
+    height: 14px;
+    padding: 0px;
+    cursor: pointer;
+    background-color: ${({ $color: $color }) => $color};
+
+    &:disabled {
+        cursor: default;
+    }
 `;
 
 interface SquareProps {
     color: string;
-    usedAs: SQUARE_USAGE;
-}
+    disabled?: boolean;
+    handleClick?: () => void;
```

Comment:
The standard naming for UI event callbacks is On<SomeAction>: OnClick, OnFocus, onBlur, etc. "OnClick" here would be better for this prop name.

Reply:
Addressed